linux/tests/sync_linux_ut/sync_linux_ut.c

```diff
@@ -0,0 +1,164 @@
+    }
+    return expected_return_val;
+}
+
+
+BEGIN_TEST_SUITE(sync_linux_unittests)
+
+TEST_SUITE_INITIALIZE(suite_init)
+{
+    g_testByTest = TEST_MUTEX_CREATE();
+    ASSERT_IS_NOT_NULL(g_testByTest);
+    ASSERT_ARE_EQUAL(int, 0, umock_c_init(on_umock_c_error));
+    REGISTER_GLOBAL_MOCK_HOOK(mock_syscall, hook_mock_syscall)
+}
+
+TEST_SUITE_CLEANUP(TestClassCleanup)
+{
+    umock_c_deinit();
+
+    TEST_MUTEX_DESTROY(g_testByTest);
+}
+
+TEST_FUNCTION_INITIALIZE(f)
+{
+    if (TEST_MUTEX_ACQUIRE(g_testByTest))
+    {
+        ASSERT_FAIL("our mutex is ABANDONED. Failure in test framework");
+    }
+
+    umock_c_reset_all_calls();
+}
+
+TEST_FUNCTION_CLEANUP(cleans)
+{
+    TEST_MUTEX_RELEASE(g_testByTest);
+}
+
+/*Tests_SRS_SYNC_LINUX_43_002: [ wait_on_address shall call syscall from sys/syscall.h with arguments SYS_futex, address, FUTEX_WAIT_PRIVATE, *compare_address, timeout_struct, NULL, NULL. ]*/
+/*Tests_SRS_SYNC_LINUX_43_003: [ wait_on_address shall return true if syscall returns 0.]*/
+
+TEST_FUNCTION(wait_on_address_calls_syscall_successfully)
+{
+    ///arrange
+    volatile_atomic int32_t var;
+    int32_t val = INT32_MAX;
+    atomic_exchange(&var, val);
+    check_timeout = true;
+    expected_timeout_ms = 100;
+    expected_return_val = 0;
+    STRICT_EXPECTED_CALL(mock_syscall(SYS_futex, (int32_t*)&var, FUTEX_WAIT_PRIVATE, val, IGNORED_ARG, NULL, 0));
+
+    ///act
+    bool return_val = wait_on_address(&var, &val, expected_timeout_ms);
+
+    ///assert
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls(), "Actual calls differ from expected calls");
+    ASSERT_IS_TRUE(return_val, "Return value is incorrect.");
+}
+
+/*Tests_SRS_SYNC_LINUX_43_002: [ wait_on_address shall call syscall from sys/syscall.h with arguments SYS_futex, address, FUTEX_WAIT_PRIVATE, *compare_address, timeout_struct, NULL, NULL. ]*/
+/*Tests_SRS_SYNC_LINUX_43_004: [ wait_on_address shall return false if syscall does not return 0.]*/
+TEST_FUNCTION(wait_on_address_calls_sycall_unsuccessfully)
+{
+    ///arrange
+    volatile_atomic int32_t var;
+    int32_t val = INT32_MAX;
+    atomic_exchange(&var, val);
+    check_timeout = true;
+    expected_timeout_ms = 100;
+    expected_return_val = -1;
+    STRICT_EXPECTED_CALL(mock_syscall(SYS_futex, (int*)&var, FUTEX_WAIT_PRIVATE, val, IGNORED_ARG, NULL, 0));
+
+    ///act
+    bool return_val = wait_on_address(&var, &val, expected_timeout_ms);
+
+    ///assert
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls(), "Actual calls differ from expected calls");
+    ASSERT_IS_FALSE(return_val, "Return value is incorrect.");
+}
+
+/*Tests_SRS_SYNC_LINUX_43_005: [ wake_by_address_all shall call syscall from sys/syscall.h with arguments SYS_futex, address, FUTEX_WAKE_PRIVATE, INT_MAX, NULL, NULL, NULL. ]*/
+TEST_FUNCTION(wake_by_address_all_calls_sycall)
+{
+    ///arrange
+    check_timeout = false;

```

Comment:
>check_timeout = false; [](start = 4, length = 22)

would be nice to set this in `TEST_FUNCTION_INITIALIZE`, then tests "opt-in" by setting to true.